hypertrace-core-graphql-service/src/main/java/org/hypertrace/core/graphql/service/DefaultGraphQlServiceConfig.java

```diff
@@ -22,6 +22,7 @@
 
   private static final String GATEWAY_SERVICE_HOST_PROPERTY = "gateway.service.host";
   private static final String GATEWAY_SERVICE_PORT_PROPERTY = "gateway.service.port";
+  private static final String GATEWAY_SERVICE_RPC_CLIENT_DEADLINE = "gateway.service.deadline";
```

Comment:
Why do we need to keep it as a mandatory field? Can we take the value from config if it is defined or take the default value?

Reply:
we do want a default, it's just a matter of where to set the default - defining it both in code and in config just makes it a bit more opaque to follow since it's an extra layer of resolution, and the config reading has to then handle the unset case.

Usually, we set defaults ini application.conf, and that's optionally overriden via helm (or env vars if run via docker)